crates/bitwarden-crypto/src/keys/master_key.rs

```diff
@@ -145,21 +145,29 @@ pub(super) fn decrypt_user_key(
             });
             user_key.decrypt_with_key(&legacy_key)?
         }
-        _ => {
+        EncString::Aes256Cbc_HmacSha256_B64 { .. } => {
             let stretched_key = SymmetricCryptoKey::Aes256CbcHmacKey(stretch_key(key)?);
             user_key.decrypt_with_key(&stretched_key)?
         }
+        EncString::Cose_Encrypt0_B64 { .. } => {
+            return Err(CryptoError::OperationNotSupported(
+                crate::error::UnsupportedOperation::EncryptionNotImplementedForKey,
+            ));
+        }
     };
 
     SymmetricCryptoKey::try_from(dec.as_mut_slice())
 }
 
 /// Generate a new random user key and encrypt it with the master key.
+///
+/// WARNING: This function should only be used with a proper cryptographic random number generator.
+/// If you do not have a good reason for using this, use [MasterKey::make_user_key] instead.
 fn make_user_key(
-    mut rng: impl rand::RngCore,
+    mut rng: impl rand::RngCore + rand::CryptoRng,
     master_key: &MasterKey,
 ) -> Result<(UserKey, EncString)> {
-    let user_key = SymmetricCryptoKey::generate(&mut rng);
+    let user_key = SymmetricCryptoKey::make_aes256_cbc_hmac_key_internal(&mut rng);
```

Comment:
Now that both this function and `make_aes256_cbc_hmac_key_internal` are using the same rng types, you can remove the `&mut`: 
```suggestion
    let user_key = SymmetricCryptoKey::make_aes256_cbc_hmac_key_internal(rng);
```